### PR TITLE
Expose `HF_HOME` in constants

### PR DESCRIPTION
**Pull request by Wauplin**
**TL;DR:** `HF_HOME` env variable is parsed but not exposed as a `huggingface_hub` constant. Let's expose it to harmonize things in other libraries.

---

I am currently making a cleanup of the environment variables in the HF ecosystem, and especially `transformers` (following up after https://github.com/huggingface/huggingface_hub/pull/1786).

I realized `transformers` and `datasets` need to know the `HF_HOME` directory to store python modules. At the moment the parsing is done individually in those libraries. Let's harmonize that by defining a `HF_HOME` constant in huggingface_hub directly -instead of lowercased `hf_cache_home`.

I am tempted to make a new `0.19.2` release once this is merged so that it can directly be integrated in transformers/datasets.

### CODE REVIEW

File: src/huggingface_hub/constants.py [hunk 1]
```diff
@@ -82,15 +82,16 @@ def _as_int(value: Optional[str]) -> Optional[int]:
...
 # default cache
 default_home = os.path.join(os.path.expanduser("~"), ".cache")
-hf_cache_home = os.path.expanduser(
+HF_HOME = os.path.expanduser(
     os.getenv(
         "HF_HOME",
         os.path.join(os.getenv("XDG_CACHE_HOME", default_home), "huggingface"),
     )
 )
+hf_cache_home = HF_HOME  # for backward compatibility. TODO: remove this in 1.0.0
```
Comment (julien-c) [inline src/huggingface_hub/constants.py:91]: 👍

**Review by julien-c — APPROVED** [review]

**Comment by Wauplin** [comment]
Thanks for the review!
